backend/kernelCI_app/views/treeCommitsHistory.py

```diff
@@ -1,155 +1,285 @@
-from rest_framework.views import APIView
-from rest_framework.response import Response
-from rest_framework import status
-from django.http import JsonResponse, HttpResponseBadRequest
-from django.db import connection
 from kernelCI_app.utils import (
-    NULL_STRINGS,
+    UNKNOWN_STRING,
     FilterParams,
     InvalidComparisonOP,
     getErrorResponseBody,
+    toIntOrDefault
 )
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.http import JsonResponse, HttpResponseBadRequest
+from django.db import connection
+from typing import Dict, List
 
 
 # TODO Move this endpoint to a function so it doesn't
 # have to be another request, it can be called from the tree details endpoint
 class TreeCommitsHistory(APIView):
     def __init__(self):
-        self.filters_options = {
-            'build': {
-                'table_alias': 'b',
-                'filters': []
+        self.commit_hashes = {}
+        self.filterParams = None
+        self.field_values = dict()
+        self.processed_builds = set()
+        self.processed_tests = set()
+
+    def setup_filters(self):
+        self.filterTestDurationMin = self.filterParams.filterTestDurationMin
+        self.filterTestDurationMax = self.filterParams.filterTestDurationMax
+        self.filterBootDurationMin = self.filterParams.filterBootDurationMin
+        self.filterBootDurationMax = self.filterParams.filterBootDurationMax
+        self.filterBuildDurationMin = self.filterParams.filterBuildDurationMin
+        self.filterBuildDurationMax = self.filterParams.filterBuildDurationMax
+        self.filterBootStatus = self.filterParams.filterBootStatus
+        self.filterTestStatus = self.filterParams.filterTestStatus
+        self.filterTreeDetailsConfigs = self.filterParams.filterConfigs
+        self.filterTreeDetailsCompiler = self.filterParams.filterCompiler
+        self.filterArchitecture = self.filterParams.filterArchitecture
+        self.filterHardware = self.filterParams.filterHardware
+        self.filterTestPath = self.filterParams.filterTestPath
+        self.filterBootPath = self.filterParams.filterBootPath
+        self.filterBuildValid = self.filterParams.filterBuildValid
+        self.filterIssues = self.filterParams.filterIssues
```

Comment:
`self.filterIssues` is not being used, the commit graph is still not reacting to issue filters